internal/app/machined/pkg/controllers/siderolink/manager.go

```diff
@@ -110,20 +115,18 @@ func parseAPIEndpoint(sideroLinkParam string) (apiEndpoint, error) {
 
 // Run implements controller.Controller interface.
 //
-//nolint:gocyclo
+//nolint:gocyclo,cyclop
 func (ctrl *ManagerController) Run(ctx context.Context, r controller.Runtime, logger *zap.Logger) error {
 	if ctrl.Cmdline == nil || ctrl.Cmdline.Get(constants.KernelParamSideroLink).First() == nil {
 		// no SideroLink command line argument, skip controller
 		return nil
 	}
 
-	s, err := smbios.GetSMBIOSInfo()
```

Comment:
we should better replace this with fetching from the resource https://github.com/siderolabs/talos/blob/cbf6dc1009ad47a2804774839e4e0301efa8ac78/pkg/machinery/resources/hardware/system_information.go#L28-L37

Reply:
yep, makes sense. will do.

Reply:
done